### PR TITLE
programmer's guide directory reorg

### LINKED ISSUE #1120: [Programming Guide] Restructure "Programming" section directory in User Guide
## Describe the issue briefly
This issue is part of [Project: Restructure the O3DE Engine Programming Guide.](https://github.com/o3de/sig-docs-community/discussions/16). This issue should be resolved first, to avoid merge conflicts with the rest of the issues.

**Action items**
- [ ] Rename "Engine Core" section to "Programming" and change folder directory path from `user-guide/engine-core` to `user-guide/programming`. Find and replace all links to the new path.

- [ ] Move `user-guide/component/develoment` section to `user-guide/programming/component`. Find and replace all links to the new path.

- [ ] Move `user-guide/gem/develoment` section to `user-guide/programming/gem`. Find and replace all links to the new path.

**Output**

A PR with these changes.

## Which page(s) / section(s) are affected?

See above. 

## Does this work have an engineering dependency? What is it?

No.

### PR DESCRIPTION
**Pull request by FiniteStateGit**
[Deploy Preview](https://deploy-preview-1147--o3deorg.netlify.app/docs/user-guide/programming/)
 
## Change summary

Rename directories:
user-guide/engine -> user-guide/programming
user-guide/gems/development -> user-guide/programming/gems
user-guide/components/development -> user-guide/programming/components

Updated links and image locations.
Updated old and new index pages to reflect changes.
Updated frontmatter to reflect changes.

Image found to be missing: https://github.com/o3de/o3de.org/issues/1146 

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

Fix #1120

### CODE REVIEW

**Review by chanmosq — APPROVED** [review]
Great changes, thanks! Can you resolve merge conflicts?

**Comment by willihay** [comment]
Adding @micronAMZN and @sptramer for visibility and the chance to comment, since this is an architectural change to a core part of the user guide.

File: content/docs/user-guide/_index.md [hunk 1]
```diff
@@ -18,10 +18,10 @@ Read about [the features provided by Open 3D Engine](/docs/welcome-guide/feature
...
-| [Components](components/) | Learn about the components that add functionality to your entities in O3DE, and how to create new components. |
+| [Components](components/) | Learn about the components that add functionality to your entities in O3DE. |
 | [Editor](editor/) |  Learn how to use the O3DE Editor. |
-| [Engine](engine/) |  Learn about core engine programming concepts in O3DE. |
-| [Gems](gems/) | Learn about the Gems that extend O3DE, and how to create new Gems.|
+| [Programming](programming/) |  Learn about core engine programming concepts in O3DE. |
+| [Gems](gems/) | Learn about the Gems that extend O3DE.|
 | [Interactivity](interactivity/) | Learn how to implement player navigation controls and other world interaction methods. |
 | [Networking](networking/) | Learn about the networking features in O3DE, offering both peer-to-peer connections and dedicated servers.  |
 | [Packaging](packaging/) | Learn how to package your O3DE-based product for release. |
```
Comment (willihay) [inline content/docs/user-guide/_index.md:27]:
```suggestion
| [Gems](gems/) | Learn about the Gems that extend O3DE.|
| [Interactivity](interactivity/) | Learn how to implement player navigation controls and other world interaction methods. |
| [Networking](networking/) | Learn about the networking features in O3DE, offering both peer-to-peer connections and dedicated servers.  |
| [Packaging](packaging/) | Learn how to package your O3DE-based product for release. |
| [Programming](programming/) |  Learn about core engine programming concepts in O3DE. |
```

File: content/docs/user-guide/programming/components/_index.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
-description: Learn how to create and reflect components in C++ using the O3DE Component Entity System.
-linktitle: Component Development
+linkTitle: Component Development
 title: Programmer's Guide to Component Development
+description: Learn how to create and reflect components in C++ using the O3DE Component Entity System.
 weight: 200
```
Comment (willihay) [inline content/docs/user-guide/programming/components/_index.md:5]: Let's make sure gem and component development appear near the top of the ToC, and in that order (to match their order of appearance in the gem and component reference sections).
```suggestion
weight: 140
```

File: content/docs/user-guide/programming/gems/_index.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ---
-title: Gem Development
+linkTitle: Gem Development
+title: Programmer's Guide to Gem Development
+description: Learn how to create your own Gems in Open 3D Engine (O3DE).
 weight: 200
```
Comment (willihay) [inline content/docs/user-guide/programming/gems/_index.md:5]: Let's make sure gem and component development appear near the top of the ToC, and in that order (to match their order of appearance in the gem and component reference sections).
```suggestion
weight: 120
```

File: content/docs/user-guide/programming/components/_index.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
-description: Learn how to create and reflect components in C++ using the O3DE Component Entity System.
-linktitle: Component Development
+linkTitle: Component Development
 title: Programmer's Guide to Component Development
+description: Learn how to create and reflect components in C++ using the O3DE Component Entity System.
```
Comment (FiniteStateGit) [inline content/docs/user-guide/programming/components/_index.md:4]:
```suggestion
description: Learn how to create and reflect components in C++ and Python using the O3DE Component Entity System.
```
Python too?

File: content/docs/user-guide/programming/remote-console.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ Universal Remote Console requires the use of a PC and works with both Android an
 
 1. After you update the `.cfg` file, deploy the game to mobile devices.
 
-   For more information, see [Deploy with O3DE Editor](/docs/userguide/mobile/android/build-deploy#run-the-deployment-tool) and [Creating a Release App](/docs/user-guide/platforms/ios/creating-release-app/).
+   For more information, see [Deploy with O3DE Editor](/docs/user-guide/mobile/android/build-deploy#run-the-deployment-tool) and [Creating a Release App](/docs/user-guide/platforms/ios/creating-release-app/).
```
Comment (willihay) [inline content/docs/user-guide/programming/remote-console.md:32]: Even though this topic is in draft, let's get the intended target link corrected, or - in the case of iOS - what one might presume to be the future correct location.
```suggestion
   For more information, see [Android support for O3DE](/docs/user-guide/platforms/android/)<!-- and [iOS support for O3DE](/docs/user-guide/platforms/ios/)-->.
```

**Comment by FiniteStateGit** [comment]
Should I ignore edits to api docs?  There are a few in here.

**Review by willihay — CHANGES_REQUESTED** [review]
For this change, we'll also want to have some redirects. Please append the following to /static/_redirects:

/docs/user-guide/engine/* /docs/user-guide/programming/:splat
/docs/user-guide/components/development/* /docs/user-guide/programming/components/:splat
/docs/user-guide/gems/development/* /docs/user-guide/programming/gems/:splat

**Comment by willihay** [comment]
> Should I ignore edits to api docs? There are a few in here.

In my opinion, I'd leave them. Yes, they would get overwritten next time the static API pages are updated. But I'd just leave the fixes in here for now and then make sure to file GitHub issues in https://github.com/o3de/o3de to get them corrected.

**Comment by FiniteStateGit** [comment]
Done. Thank you for the suggestions.

File: content/docs/user-guide/programming/components/_index.md (hunk 4 above)
Comment (willihay) [reply content/docs/user-guide/programming/components/_index.md:4]: Reflecting methods to Python are not currently covered in this section. I would leave this out.

**Review by willihay — APPROVED** [review]

File: content/docs/user-guide/programming/components/_index.md (hunk 4 above)
Comment (chanmosq) [reply content/docs/user-guide/programming/components/_index.md:4]: I know O3DE supports tools and plugins written in Python. I'm not entirely sure about O3DE components, I'll ask sig-build in Discord.

**Comment by chanmosq** [comment]
Holding from approval to verify that we can develop O3DE components using Python.

File: content/docs/user-guide/programming/components/_index.md (hunk 4 above)
Comment (chanmosq) [reply content/docs/user-guide/programming/components/_index.md:4]: Just saw @willihay's comment 🙂 - Agreed, leave Python out. 
To add to the explanation, O3DE doesn't run Python at runtime, only in-editor.

**Pull request merged**